tools/gfx/debug-layer.cpp

```diff
@@ -0,0 +1,948 @@
+#include "debug-layer.h"
+#include "renderer-shared.h"
+#include "slang-gfx.h"
+
+using namespace Slang;
+
+namespace gfx
+{
+#ifdef __FUNCSIG__
+#    define SLANG_FUNC_SIG __FUNCSIG__
+#elif defined(__PRETTY_FUNCTION__)
+#    define SLANG_FUNC_SIG __FUNCSIG__
+#elif defined(__FUNCTION__)
+#    define SLANG_FUNC_SIG __FUNCTION__
+#else
+#    define SLANG_FUNC_SIG "UnknownFunction"
+#endif
+
+thread_local const char* _currentFunctionName = nullptr;
+struct SetCurrentFuncRAII
+{
+    SetCurrentFuncRAII(const char* funcName) { _currentFunctionName = funcName; }
+    ~SetCurrentFuncRAII() { _currentFunctionName = nullptr; }
+};
+#define SLANG_GFX_API_FUNC SetCurrentFuncRAII setFuncNameRAII(SLANG_FUNC_SIG)
+#define SLANG_GFX_API_FUNC_NAME(x) SetCurrentFuncRAII setFuncNameRAII(x)
```

Comment:
Setting a thread-local variable on entry to every `gfx` function seems like it might introduce a lot of overhead. Have we confirmed that the the performance impact isn't actually a concern?